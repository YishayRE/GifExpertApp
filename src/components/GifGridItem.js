import React from 'react'
import PropTypes from 'prop-types'


const GifGridItem = ({id, title, url}) => {

    return (
        <div className="card animate__animated animate__bounce">
            <img src={url} alt={title}></img>

            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}   

export default GifGridItem