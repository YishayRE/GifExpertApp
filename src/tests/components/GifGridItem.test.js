import React from "react";
import { shallow } from "enzyme";

import GifGridItem from "../../components/GifGridItem";

describe('Pruebas en el componente GifGridItem', () => {
    const title = 'Hola';
    const url = 'url.com';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe de mostrar <GifGridItem/>', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo', () => {
        const textoP = wrapper.find('p').text().trim();

        expect(textoP).toBe(title);
    });

    test('Debe de mostrar una imagen', () => {
        const img = wrapper.find('img');
        const src = img.props().src;
        const alt = img.props().alt;

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('Debe el div tener una clase', () => {
        const clase = 'animate__bounce';
        const img = wrapper.find('div');
        const classN = img.props().className;
        expect(classN).toMatch(clase);
    });
/*
    

    test('Debe de incrementar el boton en 1', () => {
        wrapper.find('button').at(2).simulate('click');
        
        const textoP = wrapper.find('h2').text().trim();
        
        expect(textoP).toBe('11');
    });

    test('Debe de reiniciar el boton', () => {
        wrapper.find('button').at(1).simulate('click');
        
        const textoP = wrapper.find('h2').text().trim();
        
        expect(textoP).toBe('10');
    });

    test('Debe de disminuir el boton en 1', () => {
        wrapper.find('button').at(0).simulate('click');
        
        const textoP = wrapper.find('h2').text().trim();
        
        expect(textoP).toBe('9');
    });
*/
});