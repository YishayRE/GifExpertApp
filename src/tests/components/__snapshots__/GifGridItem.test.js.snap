// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe de mostrar <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Hola"
    src="url.com"
  />
  <p>
    Hola
  </p>
</div>
`;
